sdk/ai/azure-ai-projects/azure/ai/projects/aio/operations/_patch.py

```diff
@@ -0,0 +1,2472 @@
+
+
+class ConnectionsOperations(ConnectionsOperationsGenerated):
+
+    @distributed_trace_async
+    async def get_default(
+        self, *, connection_type: ConnectionType, with_credentials: bool = False, **kwargs: Any
+    ) -> ConnectionProperties:
+        """Get the properties of the default connection of a certain connection type, with or without
+        populating authentication credentials.
+
+        :param connection_type: The connection type. Required.
+        :type connection_type: ~azure.ai.projects.models._models.ConnectionType
+        :param with_credentials: Whether to populate the connection properties with authentication credentials. Optional.
+        :type with_credentials: bool
+        :return: The connection properties, or `None` if there are no connections of the specified type.
+        :rtype: ~azure.ai.projects.models._models.ConnectionProperties
+        :raises ~azure.core.exceptions.HttpResponseError:
+        """
+        kwargs.setdefault("merge_span", True)
+        if not connection_type:
+            raise ValueError("You must specify an connection type")
+        # Since there is no notion of default connection at the moment, list all connections in the category
+        # and return the first one
+        connection_properties_list = await self.list(connection_type=connection_type, **kwargs)
+        if len(connection_properties_list) > 0:
+            if with_credentials:
+                return await self.get(
+                    connection_name=connection_properties_list[0].name, with_credentials=with_credentials, **kwargs
+                )
+            else:
+                return connection_properties_list[0]
+        else:
+            return None
+
+    @distributed_trace_async
+    async def get(self, *, connection_name: str, with_credentials: bool = False, **kwargs: Any) -> ConnectionProperties:
+        """Get the properties of a single connection, given its connection name, with or without
+        populating authentication credentials.
+
+        :param connection_name: Connection Name. Required.
+        :type connection_name: str
+        :param with_credentials: Whether to populate the connection properties with authentication credentials. Optional.
+        :type with_credentials: bool
+        :return: The connection properties, or `None` if a connection with this name does not exist.
+        :rtype: ~azure.ai.projects.models._models.ConnectionProperties
+        :raises ~azure.core.exceptions.HttpResponseError:
+        """
+        kwargs.setdefault("merge_span", True)
+        if not connection_name:
+            raise ValueError("Endpoint name cannot be empty")
+        if with_credentials:
+            try:
+                connection: GetConnectionResponse = await self._get_connection_with_secrets(
+                    connection_name=connection_name, ignored="ignore", **kwargs
+                )
+            except ResourceNotFoundError as _:
+                return None
```

Comment:
Just let it raise?

Reply:
I changed it to throw the exception when the connection is not found, instead of returning None.